pure-pulsars/src/cmds/wikianimal.py

```diff
@@ -0,0 +1,151 @@
+import logging
+import os
+import re
+
+import aiohttp
+import discord
+from discord import NotFound, app_commands
+from discord.app_commands.errors import CommandInvokeError
+from pint import UnitRegistry
+
+import button_class
+from button_class import GameType, GiveUpButton, GuessButton, _Button
+from wikiutils import UA, make_img_embed, search_wikipedia
+
+UREG = UnitRegistry()
+
+WEIGHT_PATTERN = re.compile(r"(\d+ ?(?:kg|tons|lbs|oz|g|pounds|grams))")
+
+
+class IncompatibleAnimalError(Exception):
+    """Error to be thrown if wikipedia animal and/or animal api fail to provide a valid weight range."""
+
+    def __init__(self) -> None:
+        super().__init__("No valid animal weights")
+
+
+class WinLossFunctions(button_class.WinLossManagement):
+    """The Basic Win Loss Function for WikiAnimal."""
+
+    def __init__(self, winargs: dict, lossargs: dict) -> None:
+        super().__init__(winargs, lossargs)
+
+    async def on_win(self) -> None:
+        """Clean up on win."""
+
+    async def on_loss(self) -> None:
+        """Clean up on loss."""
+
+
+async def find_animal_weight(animal_name: str) -> list[str]:
+    """Determine animal's weight ranges based on article text."""
+    api_url = f"https://api.api-ninjas.com/v1/animals?name={animal_name}"
+    async with (
+        aiohttp.ClientSession(timeout=aiohttp.ClientTimeout(total=20)) as session,
```

Comment:
This is quite a long timeout, won't the discord interaction timeout by then? Is it worth reducing it?